### PR TITLE
fix version string

### PR DESCRIPTION
**Pull request by charles-cooper**
### What I did
- fix version string in `make freeze` generated binaries
- also fix the binary name in the "build.yml" workflow
- see https://gitlab.com/python-devs/importlib_metadata/-/issues/71

- remove pyproject.toml usage of setuptools_scm as it generates a  
  conflicting version string depending on build context
  (`setup.py install`, `pip install`, `make freeze`)

### How I did it
fix the case where the setuptools_scm default version string takes over

### How to verify it
```bash
$ make freeze
...
12198 INFO: Building PKG (CArchive) vyper.0.3.2+commit.c47ce229.linux.pkg completed successfully.
....
$ dist/vyper.0.3.2+commit.c47ce229.linux --version
0.3.2+commit.cf5fad74
```
also testing the artifact workflow here: https://github.com/vyperlang/vyper/actions/runs/1854199900

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: vyper/__init__.py [hunk 1]
```diff
@@ -10,4 +10,10 @@
 try:
     __version__ = _version(__name__)
 except PackageNotFoundError:
-    from vyper.version import version as __version__
+    import re
+
+    from vyper.version import version as version_str
+
+    version_str = re.sub(r"\.dev\d+", "", version_str)
```
Comment (fubuloubu) [inline vyper/__init__.py:17]: As comment "clean dirty tag"

File: vyper/__init__.py [hunk 2]
```diff
@@ -10,4 +10,10 @@
...
+    version_str = re.sub(r"\+g([a-f0-9]+).*", r"+commit.\1", version_str)
```
Comment (fubuloubu) [inline vyper/__init__.py:18]: Add comment "reformat commit string"

**Review by fubuloubu — COMMENTED** [review]
You can't use a formatter in setup.py for setuptools_scm to achieve this?

File: vyper/__init__.py (hunk 2 above)
Comment (charles-cooper) [reply vyper/__init__.py:18]:
```suggestion
    # fix commit hash
    # see `_local_version()` in setup.py
    version_str = re.sub(r"\+g([a-f0-9]+).*", r"+commit.\1", version_str)
```

**Comment by charles-cooper** [comment]
there *was* an error in our configuration. we triggered `setuptools_scm` from both https://github.com/vyperlang/vyper/blob/56a034fce93e9613643c7cee1dafe9b616f62629/setup.py and https://github.com/vyperlang/vyper/blob/56a034fce93e9613643c7cee1dafe9b616f62629/pyproject.toml which resulted in different version strings depending on the build context.

I've updated the PR so that setuptools_scm is only configured from `setup.py` (which is the recommended usage from https://github.com/pypa/setuptools_scm/issues/418#issuecomment-622835221), and gets rid of the adhoc fix for `version.py`.

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**